calico_kubernetes/calico_kubernetes.py

```diff
@@ -630,9 +639,17 @@ def _get_pod_config(self):
             path = os.path.join(self.api_root,
                                 'namespaces/%s/pods/%s' % (self.namespace,
                                                            self.pod_name))
+
+
             try:
                 logger.debug('Querying API for Pod: %s', path)
-                response = session.get(path, verify=False)
+                
+                if (self.client_certificate and self.ca_certificate):
```

Comment:
I'd suggest sticking some debug logging in these `if/else` branches so we can easily trace what is happening.